### PR TITLE
fix: [Dupe detection] Tax code review step shows empty option when workspace does not have tax rate

### PR DESCRIPTION
**Pull request by Krishna2323**
### Fixed Issues
$ https://github.com/Expensify/App/issues/47796
PROPOSAL: https://github.com/Expensify/App/issues/47796#issuecomment-2302529441

### Tests
1. Workspace A has tax rates, categories, and tags.
2. Workspace B does not have taxes, categories, and tags feature enabled.
3. Open the chat for Workspace A.
4. Submit an expense in Workspace A with a tax rate, category, and tag (use the same amount and merchant).
5. Submit another expense in Workspace A with all fields left empty (defaults).
6. Open the chat for Workspace B.
7. Submit an expense in Workspace B with the same amount and merchant as the previous entries.
8. Open the transaction thread for the expense submitted in Workspace B.
9. Click Review Duplicates.
10. Choose Keep this one for the first transaction and verify that the tax, tag, and category review step is skipped since Workspace B does not have these features enabled.
11. Return to the initial duplicate review step (where you select the transaction to keep).
12. Choose Keep this one for the second transaction and verify that the tax, tag, and category review step is not skipped, as Workspace A has these features enabled.
13. Repeat steps 11 and 12 for the third transaction, ensuring the tax, tag, and category review step is not skipped since it belongs to Workspace A.
14. Return to the chat for Workspace A and open any transaction thread for an expense submitted there.
15. Click Review Duplicates.
16. Choose Keep this one for the first transaction and verify that the tax, tag, and category review step is not skipped, as Workspace A has these features enabled.
17. Return to the initial duplicate review step.
18. Choose Keep this one for the second transaction and verify that the tax, tag, and category review step is not skipped since Workspace A has these features enabled.
19. Return to the initial duplicate review step again.
20. Choose Keep this one for the third transaction and confirm that the tax, tag, and category review step is skipped, as this transaction belongs to Workspace B, which does not have these features enabled.

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Dupe detection doesn't work in offline mode.

### QA Steps
1. Workspace A has tax rates, categories, and tags.
2. Workspace B does not have taxes, categories, and tags feature enabled.
3. Open the chat for Workspace A.
4. Submit an expense in Workspace A with a tax rate, category, and tag (use the same amount and merchant).
5. Submit another expense in Workspace A with all fields left empty (defaults).
6. Open the chat for Workspace B.
7. Submit an expense in Workspace B with the same amount and merchant as the previous entries.
8. Open the transaction thread for the expense submitted in Workspace B.
9. Click Review Duplicates.
10. Choose Keep this one for the first transaction and verify that the tax, tag, and category review step is skipped since Workspace B does not have these features enabled.
11. Return to the initial duplicate review step (where you select the transaction to keep).
12. Choose Keep this one for the second transaction and verify that the tax, tag, and category review step is not skipped, as Workspace A has these features enabled.
13. Repeat steps 11 and 12 for the third transaction, ensuring the tax, tag, and category review step is not skipped since it belongs to Workspace A.
14. Return to the chat for Workspace A and open any transaction thread for an expense submitted there.
15. Click Review Duplicates.
16. Choose Keep this one for the first transaction and verify that the tax, tag, and category review step is not skipped, as Workspace A has these features enabled.
17. Return to the initial duplicate review step.
18. Choose Keep this one for the second transaction and verify that the tax, tag, and category review step is not skipped since Workspace A has these features enabled.
19. Return to the initial duplicate review step again.
20. Choose Keep this one for the third transaction and confirm that the tax, tag, and category review step is skipped, as this transaction belongs to Workspace B, which does not have these features enabled.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/16ccdbac-e486-4c19-99fe-cbc02eec6b81

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/b2f613ba-5706-4d52-ab1b-9a75263f81c2

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/56be230a-244a-4b93-a96c-9e91eb8e58bd

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/924f98f5-d94a-45a7-b70b-036b919d2b76

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/247bc780-c4b6-4133-bc64-e19735d6c7e7

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/ffd2ed62-fa7b-4af9-b91d-d1ff07ea079d

</details>

### CODE REVIEW

**Comment by brunovjk** [comment]
@Krishna2323 I'm dealing with some problems with my internet. I'll come back here as soon as I resolve it. Thank you

**Comment by brunovjk** [comment]
I'm back @Krishna2323  :D Now at full steam. I'll review it and let you know.

**Comment by brunovjk** [comment]
> I have changed my approach a bit from my [proposal here](https://github.com/Expensify/App/issues/47796#issuecomment-2302529441).

@pecanoro I’ve tested the recent changes, and everything runs fine without errors. The original issue appears to be resolved. However, before we proceed, could you confirm if the updated behavior, as mentioned in @Krishna2323's [comment](https://github.com/Expensify/App/pull/48958#issuecomment-2354310004), aligns with our project expectations? Specifically, we need to ensure the behavior regarding the tax code selection is correct. Once you confirm, I will do a detailed code review and complete the checklist. Is it ok? Thank you!

**Comment by pecanoro** [comment]
@brunovjk @Krishna2323 Yeah, I think that's the best behavior. Thank you for handling edge cases.

**Comment by brunovjk** [comment]
> @brunovjk @Krishna2323 Yeah, I think that's the best behavior. Thank you for handling edge cases.

Awesome!! Thank you very much @pecanoro, I will continue with the checklist.

**Comment by brunovjk** [comment]
@Krishna2323 Would you mind merging main to update the branch? Thanks.

**Comment by pecanoro** [comment]
@brunovjk All yours again!

**Comment by brunovjk** [comment]
1. Workspace A has `tax rates` and `categories`.
2. Workspace B does **not** have `tax rates` and `categories` enabled.
3. Go to workspace chat (workspace A)
4. Submit an expense with a `tax rate`and a `category`
5. Go to workspace chat (workspace B)
6. Submit the same expense as Step 3 but without `tax rate` and `category`
7. Submit another expense (any amount) to workspace chat (workspace B)
8. Open transaction thread of the expense submitted in Step 5
9. Click Review duplicates
10. Click Keep this one (any)
11. Direct to "Confirm details" page 
12. Click the back button
13. Review duplicate `category` page
14. Click the back button again
15. Click Keep this one (any)
16. Direct to "Confirm details" page 
17. Click confirm "Confirm"
18. We are back to the report page, but the field `category` as an error.

@Krishna2323 can you reproduce it? 


https://github.com/user-attachments/assets/c3c77655-a674-46b2-a2f7-ca0e095f3374

**Comment by brunovjk** [comment]
1. Workspace A has `tax rates` and `categories`.
2. Workspace B does **not** have `tax rates` and `categories` enabled.
3. Go to workspace chat (workspace A)
4. Submit an expense with a `tax rate`and a `category`
5. Go to workspace chat (workspace B)
6. Submit the same expense as Step 3 but without `tax rate` and `category`
7. Submit another expense (any amount) to workspace chat (workspace B)
8. Open transaction thread of the expense submitted in Step 5
9. Click Review duplicates
10. Click Keep this one (any)
11. Direct to "Confirm details" page 
12. Click the back button
13. Review duplicate `category` page
14. Click the back button again
15. Click Keep this one (any)
16. Direct to "Confirm details" page 
17. Click confirm "Confirm"
18. We are back to the report page, but the field `category` as an error.

@Krishna2323 Is this expected? I think not, right? can you reproduce it? 


https://github.com/user-attachments/assets/c3c77655-a674-46b2-a2f7-ca0e095f3374

**Comment by brunovjk** [comment]
I'm sorry, I didn't have time to come back here this weekend, first thing in the morning I'll try to reproduce it and I tell you @Krishna2323 Thank you.

**Comment by brunovjk** [comment]
In fact @Krishna2323, I can't reproduce the [error we had before](https://github.com/Expensify/App/pull/48958#issuecomment-2369797639). I have to focus on another PR now, tomorrow I'll come back and test it thoroughly, if everything goes well I'll complete the checklist. Before, can you check this typesrcipt error? If not, I will do it tomorrow. Thank you very much for the effort :D

**Comment by brunovjk** [comment]
Very good @Krishna2323, the initial issue seems to have been successfully resolved:

https://github.com/user-attachments/assets/2420d090-25ee-43dd-87a4-c8dfb07e425c

But we still have the category and tag issues (which do not exist in workspace B). Should we update it to be the same as Tax? In other words, skip it if it does not exist in workspace B. Maybe we can try to increase the compensation (not guaranteed), if we realize that the scope was larger than expected in the issue. Please let me know what you think.

**Comment by dylanexpensify** [comment]
@Krishna2323 @brunovjk can we get this moving again now that we decided in slack the path forward?

**Comment by brunovjk** [comment]
Thanks for the work @Krishna2323, I will review it today.

**Comment by pecanoro** [comment]
Amazing! Let's try to focus on finishing this today or tomorrow if there are no major issues; there are only two issues left before we can finally finish and close this project!

**Comment by brunovjk** [comment]
I've been testing your latest changes and it seems like we've still only skipped the tax:

https://github.com/user-attachments/assets/ecef590d-1df8-49e7-8ab6-817f6d2cc873


@Krishna2323 were you able to test well after your latest changes? Maybe I'm doing something wrong. I've already pulled the last changes. Thanks.

**Comment by dylanexpensify** [comment]
Bump @Krishna2323 can we confirm?

**Comment by brunovjk** [comment]
Thank you @Krishna2323, you are the one hehehe
Please let me know if I can assist some how

**Comment by brunovjk** [comment]
> check if that if the dupe detection works with different tags (multilevel tags) as `state`, `country` etc. It would be great help for me.

@Krishna2323 Do you mean that? 

https://github.com/user-attachments/assets/1bcc665a-f33e-4a9f-aaf5-7746569e554a

I tested it on the last main, I couldn't test it on this PR due to errors. You can test multi tags like this, if you dont know:

-------
**Precondition**:
   - Workspace is under Control plan
   - Workspace is imported with dependent tags in OD [Dependent+without+GL+codes+format.csv](https://github.com/user-attachments/files/16445673/Dependent%2Bwithout%2BGL%2Bcodes%2Bformat.csv)
  
**Steps**:
- Back to ND
- Go to workspace settings > Tags
- Click on any tag list
- Verify you see a multi tag list
-------


Please let me know if I missed anything. Thank you very much.

**Comment by brunovjk** [comment]
Great @Krishna2323! Let me test it a bit and I'll get back to you soon.

**Comment by brunovjk** [comment]
Great work @Krishna2323, I've been testing and the changes are looking good. I can reproduce a bug where the tax field is hidden, but it's also reproducible in main:

Screenshot 2024-10-10 at 12 04 09

I don't think it's related to our PR but it's getting in the way of me properly testing our PR. Can you test it properly? Also on other platforms and complete your checklist? I'll complete mine next. Thanks a lot.

**Comment by brunovjk** [comment]
@Krishna2323 Sorry, I hadn't seen it. Yes, I agree with you, it may be a little out of scope.

**Comment by brunovjk** [comment]
Sorry, I expressed myself badly, I didn't mean to say that we would implement "skip" for all, only for "multi" which is not. We can try to simplify it to move forward, the important thing is to solve the [initial issue](https://github.com/Expensify/App/issues/47796), _"the suggestion to skip the field review step when no field options are available is logical and improves the flow."_ would be the perfect addition here, but following pecanoro's [comment](https://github.com/Expensify/App/pull/48958#issuecomment-2397042740), maybe we fix the initial problem only (_using `ONYXKEYS.REVIEW_DUPLICATES` instead of `route.params.threadReportID` to correctly fetch the `transactionID`_), and improve the flow later. But this is just a suggestion @Krishna2323, since you told me you only have a few hours in the day. Thoughts @pecanoro  ? Thank you :D

**Comment by pecanoro** [comment]
We should only skip the flow if the tag is the same. In the case of multilevel, we can skip if it's exactly the same (like the string representation is the same); otherwise, show it as I saw in one of the videos. Was that the question?

**Comment by brunovjk** [comment]
> We should only skip the flow if the tag is the same. In the case of multilevel, we can skip if it's exactly the same (like the string representation is the same); otherwise, show it as I saw in one of the videos. Was that the question?

What do you think @Krishna2323 ?

**Comment by brunovjk** [comment]
We still have an issue (not related to this PR) where the "Taxes" field is hidden, making it difficult to test the PR properly.

https://github.com/user-attachments/assets/06209a89-2921-4b9a-9779-ea93348fe50a

I couldn't find an issue related to this, I will report it in the "expensify-bugs" channel. Do you have the same problem @Krishna232? Or maybe is it something on my side. Thank you.

**Comment by pecanoro** [comment]
> Going back to our PR, can I test and complete the checklist, or do we need some more adjustments? Thank you very much.

@Krishna2323 Do we need more changes or is it ready for review?

**Comment by brunovjk** [comment]
> @brunovjk, you can start reviewing.

@Krishna2323 I pulled the latest changes, but I can still reproduce the issue:

After clicking `Review duplicates` in the **Workspace B** thread, the **first report** on the `Review duplicates` RHP, corresponding to the expense made in the **Workspace B Chat**, still asks to confirm the `Category`, even though **Workspace B** doesn't have any features enabled. However, for all the other fields, they are being skipped correctly. Additionally, when returning to the same `Review duplicates` RHP and selecting an expense from Workspace A, the fields that should not be skipped are working fine, except for the tax rate; the initial issue is still reproducible on my end.

https://github.com/user-attachments/assets/2b2b97b5-750b-4718-b6f4-31ad601e7159


https://github.com/user-attachments/assets/375c1324-5d3c-4ea2-b652-3de5881f1f7b

**Comment by brunovjk** [comment]
> @brunovjk, you start testing [these issues](https://github.com/Expensify/App/pull/48958#issuecomment-2420112762), I found another issue on confirmation page, trying to fix that asap.

Great!!

**Comment by pecanoro** [comment]
Any updates?

**Comment by pecanoro** [comment]
@Krishna2323 Any updates?

**Comment by brunovjk** [comment]
> We are super close 🫠, @brunovjk please let me know if you have tested [these issues](https://github.com/Expensify/App/pull/48958#issuecomment-2420112762), we have one minor bug on confirmation page and I'm on it.

Awesome! I'm on it. I'll test it out tomorrow first thing in the morning.

**Comment by brunovjk** [comment]
Great catch, I'm still testing it but everything looks good so far. I'll be back as soon as it's ready.

**Comment by brunovjk** [comment]
Great work, @Krishna2323! You're a true warrior—thank you for your persistence! 😄 I’ve tested the latest changes thoroughly and couldn't find any bugs:

<details>
<summary>MacOS: Chrome</summary>

https://github.com/user-attachments/assets/bdf66b00-9394-43ff-b990-cb3625b876c1



</details>

That said, I think it might be helpful to create a comprehensive **Test steps**, given that we’ve made several adjustments since the original issue. @pecanoro, could you assist with outlining the key test scenarios and expected results? Specifically, when comparing fields like categories, tags, and taxes between transactions from different workspaces (e.g., Workspace A with all features enabled and Workspace B with none), it would be useful to confirm how the behavior should be in such cases.

For example, I tested with three duplicate transactions: two from Workspace A (one with all fields filled and another with some fields empty) and one from Workspace B (where certain fields like tax and category weren’t available). @pecanoro  If you think this covers everything, I'll proceed with completing the checklist, but I’d appreciate your feedback on this.

**Comment by brunovjk** [comment]
Great! Filling the checklist in a moment

**Comment by brunovjk** [comment]
Almost there @Krishna2323 🫠😂 Everything works perfectly when we are in the same workspace across all the scenarios I’ve tested. However, when reviewing transactions from different workspaces, I hit an issue. Specifically, when I try to view the confirmation page for an original expense from workspace 1 while in the chat of workspace 2, I encounter a "not found" page. If I switch to the workspace 1 chat preview and review the expense directly from there, the confirmation page loads as expected.

I’ve reproduced this bug a few times and recorded a video to demonstrate it. Please let me know if anything is unclear, thanks! 

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/user-attachments/assets/30c78199-b295-4b35-9369-ab0b0c391dc9

</details>

I’m working on narrowing down the issue. I believe we’re almost there. @pecanoro, how’s our deadline looking? Appreciate the patience!

**Comment by pecanoro** [comment]
@Krishna2323 There is no hard deadline, but we want to open the beta to more domains and we are waiting on the final touches 😄

**Comment by brunovjk** [comment]
**Pre-requisites**:
- **Workspace 1**: All features enabled.
- **Workspace 2**: All features disabled.

**Submit 3** expenses with the **same value and merchant**:
1. Expense created in **Workspace 1** with all fields filled.
2. Expense created in **Workspace 1** with all fields empty.
3. Expense created in **Workspace 2**.

**Steps**:
1. Go to the expense created in **Workspace 2**.
2. Click the **Review Duplicate** button.
3. Select an expense from **Workspace 1**.
4. Continue confirming the values through the tabs.

**Issue:**
Instead of reaching the **Confirmation** page, I was redirected to a **Not Found** page.

https://github.com/user-attachments/assets/c40df7f0-2a48-4b23-ad4a-04c171e15a46


**Note**: If we only have 2 expenses, one from workspace 1 and another from 2, everything works normally.

cc: @Krishna2323

**Comment by pecanoro** [comment]
@Krishna2323 Were you able to reproduce?

**Comment by brunovjk** [comment]
> @brunovjk, I have reproduced the issue you mentioned but it is also reproducible on main also. The not here page is shown when the created expenses are not yet opened and we select them in dupe detect. Should we fix that 😅?
> 
>  dupe_detect_issue_29OCT.mp4

I think Yes, we should. Did you investigate to find  solution?
What do you think @pecanoro ?

**Comment by pecanoro** [comment]
That seems to be an unrelated bug to what we were trying to fix here. @brunovjk can you report that one in #expensify-bugs so we can get it fixed in a different issue?

**Comment by brunovjk** [comment]
Great, thanks for clarifying @pecanoro, I will report on the channel in a moment. That said, I believe we are all set here. @Krishna2323 can you update/finish your checklist? I will test it again and update mine too.

**Comment by brunovjk** [comment]
Retesting

**Comment by brunovjk** [comment]
- Create two expenses, in the same workspace with all feats active, but with all distinct fields values, exact `amount`,` date` and `merchant`.
- Review the expenses.

**Expected:** (as it is working in 'main'):
- We reviewed the `Category`, `Tag` and `Tax rate` fields:

https://github.com/user-attachments/assets/beadc73d-1785-408b-950e-4418297040f0


**Reproducible in this PR:**
- The category field is skipped even though they are expenses from different categories:

https://github.com/user-attachments/assets/c0272e11-765c-4000-a171-bd40557e2f30

cc: @Krishna2323

**Comment by brunovjk** [comment]
> That seems to be an unrelated bug to what we were trying to fix here. @brunovjk can you report that one in #expensify-bugs so we can get it fixed in a different issue?

Reported: https://expensify.slack.com/archives/C049HHMV9SM/p1730298247856019

**Comment by brunovjk** [comment]
Hi @Krishna2323, were you able to confirm [the bug](https://github.com/Expensify/App/pull/48958#issuecomment-2447241728)? How did your tests go? Any luck with the Android build? Thanks a lot! 😊

**Comment by brunovjk** [comment]
Great!! I will test it tomorrow. Did you test it in different platforms? Thank you :D

**Comment by pecanoro** [comment]
@brunovjk Are you having issues with the last platforms?

**Comment by brunovjk** [comment]
Great work, @Krishna2323! Everything looks spot on, with just one final detail I’d like to confirm: **After confirming an expense**, regardless of which one it is—say, keeping an expense from workspace 1 while previewing in workspace 2—**where should the "back" button take us?** I noticed some inconsistency in my tests, but I'm unsure of the intended behavior. What do you think? I believe we're nearly ready to approve, but I'd like to review this "back after confirming" behavior to avoid regressions. Thanks!

**Comment by brunovjk** [comment]
> @brunovjk Are you having issues with the last platforms?

No. It seems the server encountered some issues, as I didn’t receive the _"duplicate error"_ as expected. I checked `https://status.expensify.com/` and there was a down during my testing, but everything appears to be back to normal now.

The PR is great in my opinion, I think we just need to adjust [one last detail](https://github.com/Expensify/App/pull/48958#issuecomment-2455383211) :D

**Review by brunovjk — APPROVED** [review]
**Pre-requisites**:
- A small screen where the "back button" is visible.
- **Workspace 1**: All features enabled (tax rate, category, and tag).
- **Workspace 2**: All features disabled.

**Submit 2** expenses with the **same value and merchant**:
1. Expense created in **Workspace 1** with some fields filled.
2. Expense created in **Workspace 2**.

**Steps**:
1. Open the expense created in **Workspace 1**.
2. Click the **Review Duplicate** button.
3. Select an expense from **Workspace 2**.
4. Go to the **Confirmation** page and **Confirm** the expense.
5. Click the "Back button."

**Issue:**
After clicking the back button, it redirects to a **Not Found** page.

| `main` | This PR |
| --- | --- |
| <video src=https://github.com/user-attachments/assets/5a4cfdd3-58f8-48c1-913c-87130c158288 />  | <video src=https://github.com/user-attachments/assets/646bce89-a556-4aca-b1d0-adb88e73c078 />  |

**Note**: The same bug appears on `main`, so it wasn’t introduced or fixed by this PR. I couldn't find any related issues. I’ll approve the PR since the initial issue was resolved, and I leave it to you, @pecanoro, to decide whether to address this in this PR or to report it separately as a new issue. Everything else looks pretty good to me :rocket:

**Comment by pecanoro** [comment]
@brunovjk Could be that one related to this [one](https://github.com/Expensify/App/issues/51809) you reported before? Either way, if it happens on main too, let's report it as a separate bug

**Comment by brunovjk** [comment]
> @brunovjk Could be that one related to this [one](https://github.com/Expensify/App/issues/51809) you reported before? Either way, if it happens on main too, let's report it as a separate bug.

I don't think so, I'll do some more tests later and report what I find on Slack.

> With that in mind, can I go ahead a do a final review?

Yes, please 😄

**Review by pecanoro — APPROVED** [review]

File: src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx [hunk 1]
```diff
@@ -294,8 +294,10 @@ function MoneyRequestPreviewContent({
 
     const navigateToReviewFields = () => {
         const backTo = route.params.backTo;
-        const comparisonResult = TransactionUtils.compareDuplicateTransactionFields(reviewingTransactionID);
-        Transaction.setReviewDuplicatesKey({...comparisonResult.keep, duplicates, transactionID: transaction?.transactionID ?? ''});
+        Transaction.abandonReviewDuplicateTransactions();
```
Comment (pecanoro) [inline src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:297]: I approved, but now that I think about it, can you add a comment before this line explaining why we need to call `Transaction.abandonReviewDuplicateTransactions();`?

**Review by pecanoro — CHANGES_REQUESTED** [review]

**Review by pecanoro — APPROVED** [review]

**Comment by pecanoro** [comment]
🎉 Merged!! 🎉

**Comment by IuliiaHerets** [comment]
This PR is failing because of issue #52210

File: src/libs/TransactionUtils/index.ts [hunk 2]
```diff
@@ -1110,7 +1115,52 @@ function compareDuplicateTransactionFields(transactionID: string): {keep: Partia
                 } else {
                     processChanges(fieldName, transactions, keys);
                 }
-            } else if (areAllFieldsEqual(transactions, (item) => keys.map((key) => item?.[key]).join('|'))) {
+            } else if (fieldName === 'taxCode') {
```
Comment (bernhardoj) [inline src/libs/TransactionUtils/index.ts:1118]: When both duplicate tax or category or tag are not valid anymore, when resolving the dupe, the tax/category/tag becomes empty. https://github.com/Expensify/App/issues/82925

**Pull request merged**